stdlib/sqlite3/dbapi2.pyi

```diff
@@ -223,29 +224,54 @@ def adapt(obj: Any, proto: Any, alt: _T, /) -> Any | _T: ...
 def complete_statement(statement: str) -> bool: ...
 
 if sys.version_info >= (3, 12):
+    @overload
     def connect(
         database: StrOrBytesPath,
         timeout: float = ...,
         detect_types: int = ...,
         isolation_level: str | None = ...,
         check_same_thread: bool = ...,
-        factory: type[Connection] | None = ...,
         cached_statements: int = ...,
         uri: bool = ...,
+        *,
         autocommit: bool = ...,
     ) -> Connection: ...
+    @overload
+    def connect(
+        database: StrOrBytesPath,
+        timeout: float = ...,
+        detect_types: int = ...,
+        isolation_level: str | None = ...,
+        check_same_thread: bool = ...,
+        factory: type[_ConnectionT] = ...,
+        cached_statements: int = ...,
+        uri: bool = ...,
+        *,
+        autocommit: bool = ...,
+    ) -> _ConnectionT: ...
 
 else:
+    @overload
     def connect(
         database: StrOrBytesPath,
         timeout: float = ...,
         detect_types: int = ...,
         isolation_level: str | None = ...,
         check_same_thread: bool = ...,
-        factory: type[Connection] | None = ...,
         cached_statements: int = ...,
         uri: bool = ...,
     ) -> Connection: ...
+    @overload
+    def connect(
+        database: StrOrBytesPath,
+        timeout: float = ...,
+        detect_types: int = ...,
+        isolation_level: str | None = ...,
+        check_same_thread: bool = ...,
+        factory: type[_ConnectionT] = ...,
```

Comment:
I'm not sure how to fix the failure raised here--I thought that's what the `@overload` does on line 254 above.